external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/OffsetManager.java

```diff
@@ -140,10 +140,16 @@ public OffsetAndMetadata findNextCommitOffset() {
 
         OffsetAndMetadata nextCommitOffsetAndMetadata = null;
         if (found) {
-            nextCommitOffsetAndMetadata = new OffsetAndMetadata(nextCommitOffset,
-                nextCommitMsg.getMetadata(Thread.currentThread()));
+            try {
+                nextCommitOffsetAndMetadata = new OffsetAndMetadata(nextCommitOffset,
+                    JSON_MAPPER.writeValueAsString(new KafkaSpout.Info(Thread.currentThread(), context)));
```

Comment:
Consider moving the JSON serialization into KafkaSpout commitOffsets method instead, then you can pass it as a parameter to this method and we can get away with having only one ObjectMapper. Currently we can just create the info once (e.g. in open) and reuse rather than creating and serializing a new one for each commit and partition.

Reply:
I think that approach is less efficient. Currently we only create and serialize the commit metadata when there are offsets ready to commit. Your proposed solution will always serialize, regardless if there are offsets ready to commit or not. Furthermore, ObjectMapper is a public static field, which means that per JVM we will be creating only two of them, on for the spout, and another for OffsetManager. We can even use the same instance on both classes if we want to.

Reply:
I think maybe I was being a little unclear. I'm saying that right now the information contained in the Info object is constant for the lifetime of the spout. We should be able to just create one Info object in KafkaSpout.open and reuse it. This means a single serialization for the lifetime of the spout, which is more efficient than doing it on every commit.

Reply:
Will do.